demo/chbench/dc.sh

```diff
@@ -298,11 +299,35 @@ dc_is_running() {
     ( dc_chbench_containers | grep -E "chbench_${1}" ) || true
 }
 
+dc_run_query() {
+    local query=$1
+    # shellcheck disable=SC2059
+    (printf "$query\n" | docker-compose run cli psql -q -h materialized -p 6875 -d materialize) || true
+}
+
+dc_check_query() {
+    local query=$1
+    local timeout=$2
+    for i in $(seq 0 "$timeout"); do
+        local result
+        result=$(dc_run_query "$query")
+        if [[ -z $result ]]; then
+            sleep 1
+        else
+            printf %s "$result"
+            return
+        fi
+    done
+    uw "query failed after $timeout attempts: $query"
+    exit 1
+}
+
+
 dc_ensure_stays_up() {
     local container=$1
     local seconds="${2-5}"
     echo -n "ensuring $container is staying up "
-    for i in $(seq "$seconds" 1); do
+    for i in $(seq 1 "$seconds"); do
```

Comment:
huh this worked for me and caused a countdown:

```
bash -c 'for i in $(seq 5 1); do echo $i ; done'
5
4
3
2
1
```

Reply:
Are you on macOS ? Could be a mac/GNU difference. On Ubuntu 19.10:

```
~/c/materialize ❯❯❯ which seq
/usr/bin/seq
~/c/materialize ❯❯❯ seq 5 1
~/c/materialize ❯❯❯ seq 1 5
1
2
3
4
5
~/c/materialize ❯❯❯ /usr/bin/seq 5 1
```

Reply:
Just use the bash builtin:

```
echo {5..1}
5 4 3 2 1
```

Reply:
the bash builtin doesn't work with variables, and I couldn't figure out how to make it:

```
$ bash -c 'v=4; for i in {$v..1}; do echo $i ; done'
{4..1}
```

Reply:
ah, need to give it the incr in gnu:

```console
ubuntu@ip-172-31-15-143:~$ bash -c 'for i in $(seq 5 -1 1); do echo $i ; done'
5
4
3
2
1
ubuntu@ip-172-31-15-143:~$ bash -c 'for i in $(seq 5 1); do echo $i ; done'
ubuntu@ip-172-31-15-143:~$
```